### PR TITLE
Fixed #32271 -- Improved consistency of docs CBV examples.

### PR DESCRIPTION
**Pull request by cpina**
https://code.djangoproject.com/ticket/32271

### CODE REVIEW

**Comment by smithdc1** [comment]
Hi @cpina :wave: 

I wonder if there are a few more examples of this? 

[Mixins](https://docs.djangoproject.com/en/3.1/topics/class-based-views/mixins/). For example, should this be `PublisherDetailView`?

https://github.com/django/django/blob/4794a0e35f12c699ef4af29b355c68ca66f425aa/docs/topics/class-based-views/mixins.txt#L316

[Generic display views](https://docs.djangoproject.com/en/3.1/topics/class-based-views/generic-display/). For example, should this be `PublisherListView`

https://github.com/django/django/blob/4794a0e35f12c699ef4af29b355c68ca66f425aa/docs/topics/class-based-views/generic-display.txt#L113

I think there is one example in `ref/class-based-views/base.txt`

https://github.com/django/django/blob/4794a0e35f12c699ef4af29b355c68ca66f425aa/docs/ref/class-based-views/base.txt#L227

Also, could you rename the commit message to past tense? (Improve -> Improved)

**Comment by cpina** [comment]
Hey @smithdc1 :wave: 

I'll make the suggested changes and revise properly the mixins and form handling sections (https://docs.djangoproject.com/en/3.1/topics/class-based-views/) as well. Thanks!

**Review by carltongibson — COMMENTED** [review]
Hey @cpina — looking good. 👍 

 Please squash when you're ready, and note the adjusted PR title for a hint to the required [commit message format](https://docs.djangoproject.com/en/dev/internals/contributing/committing-code/#committing-guidelines).

**Comment by cpina** [comment]
I changed all (I think!) the CBV naming that are needed. Some are from the tutorial. Tomorrow I'll follow the tutorial and make sure that I haven't made any mistake there, review, squash and change the status on the ticket.

**Comment by cpina** [comment]
I've pushed the latest changes, I've verified that at least what's changed is consistently changed. 

Commits are squashed.

I've followed the "polls" tutorial mechanically and the CBV naming in views, urls, text seems to match all good. I've found an error in the tutorial though but not related to this, it will be in a different PR/ticket.

Thanks for reviewing, not very fun to write or review but it feels better to be a bit more consistent!

File: docs/intro/tutorial04.txt [hunk 1]
```diff
@@ -289,7 +289,7 @@ views and use Django's generic views instead. To do so, open the
         template_name = 'polls/detail.html'
 
 
-    class ResultsView(generic.DetailView):
+    class ResultsDetailView(generic.DetailView):
```
Comment (felixxm) [inline docs/intro/tutorial04.txt:292]: IMO changes in tutorials are misleading and should be reverted them, e.g.
`IndexListView` is a main (index) view not a list of indexes, `ResultsView` is a view with a question results not details of a result.

I don't see any point in changing the current names in tutorials.

File: docs/ref/class-based-views/base.txt [hunk 2]
```diff
@@ -141,7 +141,7 @@ MRO is an acronym for Method Resolution Order.
 
         from articles.models import Article
 
-        class HomePageView(TemplateView):
+        class HomePageTemplateView(TemplateView):
```
Comment (felixxm) [inline docs/ref/class-based-views/base.txt:144]: I would revert this. We shouldn't be doing this mechanically.

File: docs/topics/class-based-views/index.txt [hunk 3]
```diff
@@ -65,7 +65,7 @@ override the template name::
     # some_app/views.py
     from django.views.generic import TemplateView
 
-    class AboutView(TemplateView):
+    class AboutTemplateView(TemplateView):
```
Comment (felixxm) [inline docs/topics/class-based-views/index.txt:68]: I would revert adding `Template` in all cases.

**Comment by cpina** [comment]
Thanks @felixxm for your comments - I see what you say, I'll revise other changes with this angle and make adjustments.

**Comment by cpina** [comment]
> Thanks @felixxm for your comments - I see what you say, I'll revise other changes with this angle and make adjustments.

I'll close this PR - I'll not have enough bandwidth for some time, I'll come back here later.

Meanwhile: if anyone else wants to take it please do so.

**Pull request closed without merging**